src/ng/compile.js

```diff
@@ -2386,8 +2412,20 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
             removeControllerBindingWatches =
               initializeDirectiveBindings(controllerScope, attrs, controller.instance, bindings, controllerDirective);
           }
+
+          if (isObject(controllerDirective.require) && !isArray(controllerDirective.require)) {
+            var controllerObj = getControllers(name, controllerDirective.require, $element, elementControllers);
+            extend(controller.instance, controllerObj);
```

Comment:
I didn't like to add parameters to `$onInit`. If we don't want to automatically bind then, instead, we should be injecting things into the constructor of the controller, like in ng2... But that is done differently for non-ancestor components